docker_cli/setup_commands.py

```diff
@@ -0,0 +1,423 @@
+    else:
+        reader = url_reader  # type: ignore
+    cuda_gen: CUDADockerfileGenerator = get_cuda_dockerfile_generator(
+        pkg_mgr=package_mgr, url_reader=reader
+    )
+    body = cuda_gen.generate_dev_dockerfile(
+        cuda_ver_major=cuda_ver_major, cuda_ver_minor=cuda_ver_minor
+    )
+
+    dockerfile = f"FROM {base}\n\n{init_lines}\n\n{body}"
+
+    prefix = universal_tag_prefix()
+    img_tag = tag if tag.startswith(prefix) else f"{prefix}-{tag}"
+
+    return Image.build(tag=img_tag, dockerfile_string=dockerfile, no_cache=no_cache)
+
+
+def setup_conda_runtime(
+    base: str, tag: str, no_cache: bool, env_file: Union[str, os.PathLike[str]]
+) -> Image:
+    """
+    Builds the Conda runtime environment image with micromamba.
+
+    Parameters
+    ----------
+    base : str
+        The name of the image upon this one will be based.
+    tag : str
+        The tag of the image to be built.
+    no_cache : bool
+        Run Docker build with no cache if True.
+    env_file : str
+        The location of the runtime environment spec-file.
+
+    Returns
+    -------
+    Image
+        The generated image.
+    """
+
+    header, body = mamba_install_dockerfile(env_specfile=env_file)
+    full_dockerfile = f"{header}\n\nFROM {base}\n\n{body}"
+
+    prefix = universal_tag_prefix()
+    img_tag = tag if tag.startswith(prefix) else f"{prefix}-{tag}"
+
+    return Image.build(
+        tag=img_tag, dockerfile_string=full_dockerfile, no_cache=no_cache
+    )
+
+
+def setup_conda_dev(
+    base: str, tag: str, no_cache: bool, env_file: Union[str, os.PathLike[str]]
+) -> Image:
+    """
+    Set up the development environment.
+
+    Parameters
+    ----------
+    base : str
+        The name of the image upon this one will be based.
+    tag : str
+        The tag of the image to be built.
+    no_cache : bool
+        Run Docker build with no cache if True.
+    env_file : str
+        The location of the runtime environment spec-file.
+
+    Returns
+    -------
+    Image
+        The generated image.
+    """
+
+    body = mamba_add_specs_dockerfile(env_specfile=env_file)
+
+    prefix = universal_tag_prefix()
+    img_tag = tag if tag.startswith(prefix) else f"{prefix}-{tag}"
+
+    full_dockerfile = f"FROM {base}\n\n{body}"
+
+    return Image.build(
+        tag=img_tag, dockerfile_string=full_dockerfile, no_cache=no_cache
+    )
+
+
+def setup_conda_add(
+    base: str,
+    tag: str,
+    no_cache: bool,
+    packages: Iterable[str],
+    channels: Optional[Iterable[str]],
+) -> Image:
+    """
+    Create an image that adds packages to another image's micromamba environment.
+
+    Parameters
+    ----------
+    base : str
+        The name of the image upon this one will be based.
+    tag : str
+        The tag of the image to be built.
+    no_cache : bool
+        Run Docker build with no cache if True.
+    packages : Iterable[str]
+        The packages to be added.
+    channels: Iterable[str], optional
```

Comment:
This argument isn't currently optional

```suggestion
    channels: Iterable[str]
```